clients/drcachesim/common/trace_entry.h

```diff
@@ -906,6 +909,16 @@ typedef enum {
      * The initial part can be used by a simulator for warmup.
      */
     OFFLINE_FILE_TYPE_BIMODAL_FILTERED_WARMUP = 0x2000,
+    /**
+     * Indicates an offline trace that contains trace templates for some system calls.
+     * The individual traces are separated by a #TRACE_MARKER_TYPE_SYSCALL marker which
+     * also specifies what system call the following trace belongs to. This file can be
```

Comment:
I agree a zipfile would be nicer. I already had a TODO elsewhere. Now that I changed the interface to accept a record_reader_t, it can be support more easily. But the zipfile support will have to be separate from this PR as I think there's already other contributors working on adding zipfile reading support for trace_entry_t.

Reply:
Also added a TODO here

Reply:
Will look at new interface.  But I am not a fan of having all the syscalls concatenated into one file: if it's not a zip file separating them, I would want them to be in separate files and we pass in a directory or list of files or something.  Dealing with a single concatenated file does not sound fun.

Reply:
IMHO it would be like we took the separate thread files for a trace and concatenated them together to make one file: it just seems to make everything dealing with the data awkward.  Those are large; could syscall traces get quite large too and we would not want to read an entire one into memory?  Even if just a few MB: there are hundreds of syscalls.  That is another argument for separate files/components, so we can stream them.